bin/NNMass.cc

```diff
@@ -0,0 +1,201 @@
+#include "lwtnn/LightweightGraph.hh"
+#include "lwtnn/parse_json.hh"
+#include <fstream>
+
+#include "TFile.h"
+#include "TH1F.h"
+#include "TTree.h"
+#include "TVector2.h"
+#include "Math/LorentzVector.h"
+#include "Math/PtEtaPhiM4D.h"
+#include "Math/PxPyPzM4D.h"
+#include "Math/Vector4Dfwd.h"
+
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/lexical_cast.hpp>
+#include <boost/program_options.hpp>
+#include <boost/regex.hpp>
+
+#include <iostream>
+
+using boost::starts_with;
+namespace po = boost::program_options;
+
+std::string folder_to_channel(std::string foldername) {
+  std::vector<std::string> foldername_split;
+  boost::split(foldername_split, foldername, [](char c) { return c == '_'; });
+  return foldername_split.at(0);
+}
+
+std::string filename_from_inputpath(std::string input) {
+  std::vector<std::string> path_split;
+  boost::split(path_split, input, [](char c) { return c == '/'; });
+  std::string filename = path_split.at(path_split.size() - 1);
+  boost::replace_all(filename, ".root", "");
+  return filename;
+}
+
+std::string outputname_from_settings(std::string input, std::string folder,
+                                     unsigned int first_entry,
+                                     unsigned int last_entry) {
+  std::string filename = filename_from_inputpath(input);
+  std::string outputname = filename + "/" + filename + "_" + folder + "_" +
+                           std::to_string(first_entry) + "_" +
+                           std::to_string(last_entry) + ".root";
+  return outputname;
+}
+
+int main(int argc, char **argv) {
+  std::string input = "output.root";
+  std::string folder = "mt_nominal";
+  std::string tree = "ntuple";
+  std::string lwtnn_config = "model.json";
+  unsigned int first_entry = 0;
+  unsigned int last_entry = 9;
+  po::variables_map vm;
+  po::options_description config("configuration");
+  config.add_options()("input",
+                       po::value<std::string>(&input)->default_value(input))(
+      "folder", po::value<std::string>(&folder)->default_value(folder))(
+      "tree", po::value<std::string>(&tree)->default_value(tree))(
+      "first_entry",
+      po::value<unsigned int>(&first_entry)->default_value(first_entry))(
+      "last_entry",
+      po::value<unsigned int>(&last_entry)->default_value(last_entry))(
+      "lwtnn_config", po::value<std::string>(&lwtnn_config)->default_value(lwtnn_config));
```

Comment:
This option won't be set by the [job_management.py](https://github.com/KIT-CMS/friend-tree-producer/blob/master/scripts/job_management.py) script. We could implement that as an additional option there in case the NNMass executable is there, but I would prefer to avoid executable specific options as good as possible.

Could we just use the default path instead (would be then hard-coded, I know...)

Reply:
Does this work?

Reply:
I'd put the `model.json` into the [data](https://github.com/KIT-CMS/friend-tree-producer/tree/master/data) folder (could be another subfolder) and then hard-code the path in the executable :D

Reply:
In that way, we wouldn't need that option and can stay with the general set of options handled by the job submission. Is the json big?

Reply:
We can keep to option for local testing and set the default so that it works for the job submission!

Reply:
The model has 22 MB, so we should not commit it in the repo.

Reply:
Hm ok. We could make a download via url to the data folder then in the checkout script & put the default to that path.

In case of job submission, it shouldn't be a problem at least for now, since we anyhow use the local setup of batch systems with local mounts. So we only have to make sure, that the data folder path is accessible in the executable.

But OK, let's postpone it to later time for the time-being.